src/main/java/com/demo/classroom/Security/Filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,97 @@
+package com.demo.classroom.Security.Filter;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.demo.classroom.Security.Service.JwtService;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter{
+
+    private final JwtService jwtService;
+    private final UserDetailsService userDetailsService;
+    
+
+    @Override
+    protected void doFilterInternal(
+            @NotNull HttpServletRequest request,
+            @NotNull HttpServletResponse response,
+            @NotNull FilterChain filterChain
+    ) throws ServletException, IOException {
+
+        final String authHeader = request.getHeader("Authorization");
+        final String refreshToken = request.getHeader("Refresh-Token");
+
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        try {
+            final String jwt = authHeader.substring(7); //"Bearer "=7
```

Comment:
The use of 7 to skip "Bearer " is a magic number. Consider defining a constant for the prefix length or using StringUtils.removeStart or similar utility methods.
```suggestion
            final String jwt = authHeader.replace("Bearer ", "");
```

Reply:
Best to use this: `final String jwt = authHeader.substring(7);`

 Because the below one: ` final String jwt = authHeader.replace("Bearer ", "");` has some problems:
 1. If the `Authorization` header contains the `"Bearer "` string more than once, the replace method will remove all occurrences, which can lead to unexpected results.
 2. If there is a case variation, such as `"BEARER " or "bearer "`, the token extraction will fail.